packages/brand-positioning-block/src/components/BlockInjectButton.tsx

```diff
@@ -0,0 +1,164 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import {
+    ActionMenu,
+    Flyout,
+    IconArrowCircleUp20,
+    IconImageStack20,
+    LoadingCircle,
+    MenuItemContentSize,
+} from '@frontify/fondue';
+import { joinClassNames } from '@frontify/guideline-blocks-shared';
+import React, { useEffect, useRef, useState } from 'react';
+import { BlockInjectButtonProps } from '../types';
+
+const BlockInjectButton = ({
+    onClick,
+    onDrop,
+    label,
+    icon,
+    secondaryLabel,
+    isLoading,
+    onAssetChooseClick,
+    onUploadClick,
+}: BlockInjectButtonProps) => {
+    const [isDraggingOver, setIsDraggingOver] = useState(false);
+    const [menuPosition, setMenuPosition] = useState<[number, number] | undefined>();
+    const buttonRef = useRef<HTMLButtonElement>(null);
+    useEffect(() => {
+        const handleClickOutside = (event: MouseEvent) => {
+            if (buttonRef.current && !buttonRef.current.contains(event.target as HTMLElement)) {
+                setMenuPosition(undefined);
+            }
+        };
+        document.addEventListener('mousedown', handleClickOutside);
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, []);
+
+    const handleDrop: React.DragEventHandler<HTMLButtonElement> = (e) => {
+        e.preventDefault();
```

Comment:
```suggestion
    const handleDrop: DragEventHandler<HTMLButtonElement> = (event) => {
        event.preventDefault();
```
Let's still keep the code guideline in mind while doing this exercise ;)

Reply:
Fixed on all occasions.

Reply:
fixed